core/src/main/java/com/sequenceiq/cloudbreak/service/parcel/ParcelService.java

```
@@ -88,11 +94,22 @@ public ParcelOperationStatus removeUnusedParcelComponents(Stack stack) throws Cl
 
     public ParcelOperationStatus removeUnusedParcelComponents(Stack stack, Set<ClusterComponent> clusterComponentsByBlueprint) throws CloudbreakException {
         LOGGER.debug("Starting to remove unused parcels from the cluster.");
-        ParcelOperationStatus removalStatus = clusterApiConnectors.getConnector(stack).removeUnusedParcels(clusterComponentsByBlueprint);
+        Set<ClouderaManagerProduct> productsFromImage = getProductsFromImage(stack);
+        ParcelOperationStatus removalStatus = clusterApiConnectors.getConnector(stack).removeUnusedParcels(clusterComponentsByBlueprint, productsFromImage);
         clusterComponentUpdater.removeUnusedCdhProductsFromClusterComponents(stack.getCluster().getId(), clusterComponentsByBlueprint, removalStatus);
         return removalStatus;
     }
 
+    private Set<ClouderaManagerProduct> getProductsFromImage(Stack stack) {
+        try {
+            StatedImage currentImage = imageService.getCurrentImage(stack.getId());
+            return clouderaManagerProductTransformer.transform(currentImage.getImage(), true, true);
+        } catch (CloudbreakImageNotFoundException | CloudbreakImageCatalogException e) {
+            LOGGER.warn("Failed to get the current image for the cluster.", e);
+            return Collections.emptySet();
+        }
```

ignoring this error might not be a good idea